lab/manifests/system/base/ironic/internal/base/init-credentials-job.yaml

```diff
@@ -38,24 +38,40 @@ spec:
     spec:
       restartPolicy: Never
       serviceAccountName: ironic-credentials
+      volumes:
+        - name: credentials
+          emptyDir: {}
       containers:
         - name: init-credentials
-          image: alpine/k8s:1.28.4
-          command: [sh, -c]
+          image: httpd:2.4.46
+          volumeMounts:
+            - name: credentials
+              mountPath: /credentials
+          command: [bash, -c]
           args:
             - |
-              if kubectl get secret ironic-credentials -o name; then
-                exit 0
-              fi
-
-              apk add apache2-utils
-
+              set -e
               username=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 16)
               password=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 64)
 
-              mkdir -p /credentials
-              echo -n "$username" > /credentials/username
-              echo -n "$password" >  /credentials/password
-              htpasswd -b -c -B /credentials/htpasswd "$username" "$password"
+              mkdir -p /credentials/tmp
+              echo -n "$username" > /credentials/tmp/username
+              echo -n "$password" >  /credentials/tmp/password
+              htpasswd -b -c -B /credentials/tmp/htpasswd "$username" "$password"
+
+              mv /credentials/tmp/* /credentials
+        - name: create-secret
+          image: bitnami/kubectl:1.28.4
+          volumeMounts:
+            - name: credentials
+              mountPath: /credentials
+          command: [bash, -c]
+          args:
+            - |
+              set -eEuo pipefail
+              if kubectl get secret ironic-credentials -o name; then
+                exit 0
+              fi
 
+              timeout 60s bash -c 'until [ -f /credentials/username ] && [ -f /credentials/password ] && [ -f /credentials/htpasswd ]; do sleep 1; done'
```

Comment:
Just move `init-credentials` into `initContainers` and you'll not need to wait here and move files from /credentials/tmp/ to /credentials/